source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,387 @@
+
+1.  Within a <Routes>, every <Route> and <Link> is relative. This results in "Route path" and "Link to" code that is more predictable and lighter.
+    
+2.  Routes are chosen based on the best match instead of being traversed in order. This avoids bugs due to unreachable routes because they were defined later in your Switch component.
+    
+3.  Routes may be nested in one place instead of being spread out in different react components.
+    
+
+### Link and NavLink
+
+The **Link** and **NavLink** components in React Router Dom are used to create links between different routes in your application. They work similarly to the element in HTML, but they use the **to** prop to specify the target route instead of the **href** attribute.
+
+Here's an example of how you might use the **Link** component:
+
+    import { Link } from 'react-router-dom';
+    
+    function Navbar() {
+      return (
+        <nav>
+          {/* This link will navigate to the "/" route when clicked */}
+          <Link to="/">Home</Link>
+    
+          {/* This link will navigate to the "/about" route when clicked */}
+          <Link to="/about">About</Link>
+    
+          {/* This link will navigate to the "/contact" route when clicked */}
+          <Link to="/contact">Contact</Link>
+        </nav>
+      );
+    }
+    
+
+The **NavLink** component is similar to the **Link** component but has additional props that you can use to style the active link differently. For example, you can use the **activeClassName** prop to specify a class name that will be applied to the **NavLink** element when the route is active. Here's an example of how you might use the **NavLink** component:
+
+    import { NavLink } from 'react-router-dom';
+    
+    function Navbar() {
+      return (
+        <nav>
+          {/* This link will navigate to the "/" route when clicked, and the "active" class will be applied when the route is active */}
+          <NavLink to="/" activeClassName="active">Home</NavLink>
+    
+          {/* This link will navigate to the "/about" route when clicked, and the "active" class will be applied when the route is active */}
+          <NavLink to="/about" activeClassName="active">About</NavLink>
+    
+          {/* This link will navigate to the "/contact" route when clicked, and the "active" class will be applied when the route is active */}
+          <NavLink to="/contact" activeClassName="active">Contact</NavLink>
+        </nav>
+      );
+    }
+    
+
+### Nested Routing
+
+### Hooks in React Router Dom
+
+**React router component library** provides a set of Hooks that allow functional components to access routing-related information and perform routing actions.
+
+Here is a list of the hooks that are available in react-router-dom version 6:
+
+**useHistory**: This hook returns an object containing methods for navigating to different routes in the application, such as push, replace, and goBack.
```

Comment:
This list of could be added to the list of <li> elements. You can use UL, OL in markdown.